src/inputs/input-base.ts

```diff
@@ -162,7 +162,7 @@ export class InputBase implements OnInit, OnChanges, DoCheck,
         const { value } = this.state.getState();
 
         if (this.canTestRegex(this.config)) {
-            if (!new RegExp(this.config.pattern as string).test(value)) {
+            if (!new RegExp(this.config.pattern as string).test((value !== null && value !== false) ? value : ''))  {
```

Comment:
It should be `value != null && value !== false` because `undefined` is convert to string too (`value != null` check `null` and and `undefinedp ). And I think it would be clearest without brkackets :+1:

Reply:
Since it's Typescript, can we just do `.test(value ? value : '')` ?

Reply:
I guess, not because it could be `0`.